sphinxcontrib/chapeldomain/chapel.py

```diff
@@ -109,7 +117,18 @@ class ChapelLexer(RegexLexer):
             (r'[a-zA-Z_][\w$]*', Name.Class, '#pop'),
         ],
         'procname': [
-            (r'([a-zA-Z_][.\w$]*|\~[a-zA-Z_][.\w$]*|[+*/!~%<>=&^|\-]{1,2})',
+            (r'([a-zA-Z_][.\w$]*|'    # regular function name, including 2ndary
+             r'\~[a-zA-Z_][.\w$]*|'   # support for legacy destructors?
+             r'[+*/!~%<>=&^|\-]{1,2})',  # operators
```

Comment:
On a positive note, thank you so much for commenting these, this is really helpful and will make it so much easier to understand going forward